session/pingpong/consumer_balance_tracker.go

```diff
@@ -214,34 +216,36 @@ func (cbt *ConsumerBalanceTracker) handleStopEvent() {
 	})
 }
 
-func (cbt *ConsumerBalanceTracker) increaseBalance(id identity.Identity, b uint64) {
+func (cbt *ConsumerBalanceTracker) increaseBCBalance(id identity.Identity, diff uint64) {
 	cbt.balancesLock.Lock()
 	defer cbt.balancesLock.Unlock()
 
 	var before uint64
 	if v, ok := cbt.balances[id]; ok {
-		before = v
+		before = v.GetBalance()
+		v.BCBalance = safeAdd(v.BCBalance, diff)
+		cbt.balances[id] = v
+	} else {
+		cbt.ForceBalanceUpdate(id)
 	}
 
-	current := safeAdd(before, b)
-	cbt.balances[id] = current
-
-	go cbt.publishChangeEvent(id, before, current)
+	go cbt.publishChangeEvent(id, before, cbt.balances[id].GetBalance())
```

Comment:
Maybe publish full struct as it changes

Reply:
I'll break a ton of users if i do that, so no, it's staying here :)